File: Data Structures & Algorithms/sort-an-array/submission-1.java
class Solution {
    public int[] sortArray(int[] nums) {
        //10 9 1 1 1 2 3 1
        // l r
        //9 10 1 1 1 2 3 1
        // l.  r
        //1 10 9 1 1 2 3 1
        //. l. r
        //1 9 10 1 1 2 3 1
        //  l    r
        //1 1 10 9 1 2 3 1
        //.   l  r
        for(int l = 0; l < nums.length-1; l++){
            for(int r = l + 1; r < nums.length; r++){
                if(nums[l] > nums[r]){
                    int temp = nums[r];
                    nums[r] = nums[l];
                    nums[l] = temp;
                }
            }
        }
        return nums;
    }
}

File: Data Structures & Algorithms/subsets-ii/submission-1.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        res = []
        stack = []
        nums.sort()
        #
        def backtrack(i: int):
            nonlocal res, stack
            if i >= len(nums):
                res.append(stack[:])
                return
            
            stack.append(nums[i])
            backtrack(i+1)
            stack.pop()
            i += 1
            while i < len(nums) and nums[i] == nums[i-1]:
                i += 1
            backtrack(i)

        backtrack(0)
        return res


